GitHub Actions/haal-centraal-npm-beheer.md

```diff
@@ -160,6 +160,11 @@ Wil je met één statement alle packages updaten naar de meest recente versie, d
 - `ncu -u`. Met deze statement worden de npm package referenties in een package.json bestand ge-update naar de meest recente versie
 - `npm install`. Met deze statement worden de packages in de node_modules bestand geupdate aan de hand van het package.json bestand
 
+## Commit naar upstream repository
+
+Tijdens het hierboven beschreven proces wordt de folder 'node_modules' lokaal gegenereerd. Deze folder moet NIET gecommit worden naar de upstream repository. Committen naar de eigen fork van die repository wordt daarmee ook afgeraden omdat het dan erg lastig wordt de folder niet naar de upstream repository te committen.
+Je kunt voorkomen dat de betreffende folder gecommit wordt door hem in het '.gitignore' bestand te definiëren of hem simpelweg te verwijderen.
+
 [1]: https://stoplight.io/open-source/spectral/
```

Comment:
Melvin heeft toch een andere methode beschreven om te voorkomen dat onderdelen van je lokale repository meegenomen worden in de push ?

Misschien moeten we die variant hier ook beschrijven, of heb ik dat verkeerd begrepen ?

Reply:
De andere methode (het gebruiken van de .git/info/exclude file) is eerder complementair dan een andere methode. Waar je definieer welke bestanden of mappen git moet ignoren, hangt af van wat deze bestanden genereert. Als het bestanden zijn die horen bij de tooling die in het project worden gebruikt of worden gegenereerd door de tooling, dan zet je de rules in .gitignore. Hiermee voorkom je ook dat andere contributors per ongeluk deze bestanden in de repo commit.
Als het bestanden zijn die je zelf toevoegt (config bestanden, notities), dan wil je die niet toevoegen aan de .gitignore omdat hiermee het bestand wordt vervuild. De ignore rules van je eigen bestanden moet je daarom in de .git/info/exclude file toevoegen. Ze gelden dan voor jouw lokale repo.

Verder lees ik nu `of hem simpelweg te verwijderen`. Dat is geen goed idee, want als je het verwijdert voordat je hebt gecommit, dan werkt het resolven en linten van het openapi.yaml bestand tijdens een commit niet meer.

Reply:
Hebben we het gebruik van het .ignore en de .git/info/exclude ergens gedocumenteerd?

Reply:
Ik heb het niet gedocumenteerd. Mijn vraag is hoeveel je hierover wil documenteren. Je kan nl. veel documentatie online vinden. Zie bijv. https://git-scm.com/docs/gitignore. Je kan mogelijk ook hier naar verwijzen. Je kan natuurlijk wel beschrijven hoe je binnen een organisatie wil werken, maar persoonlijk denk ik dat je het per project moet doen omdat het per type project kan verschillen. Ter indicatie, als je in de [gitignore repo](https://github.com/github/gitignore) kijk, dan zie je een overzicht van gitignore templates bijgehouden door GitHub

Reply:
Een verwijzing opnemen is wat mij betreft voldoende. Liever niet zelf nog eens beschrijven tenzij dat wat beschreven staat niet duidelijk genoeg is. Bedankt voor de link.